File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int low = 0;
        int high = height.length - 1;
        int max = 0;

        while (low < high) {
            int h = Math.min(height[low], height[high]);
            int width = high - low;
            int area = width * h;

            max = Math.max(max, area);

            if (height[low] <= height[high]) {
                low++;
            } else {
                high--;
            }
        }
        return max;

    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int low = 0;
        int high = height.size() -1;
        int maxValue = 0;

        while(low < high){
            int width = high - low;
            int area = width * min(height[low], height[high]); 

            maxValue = max(area, maxValue);

            if(height[low] < height[high]){
                low++;
            }else{
                high--;
            }
        }
        return maxValue;
    }
};